compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1215,7 +1216,111 @@ pub fn prohibit_assoc_item_binding(
         } else {
             None
         },
-    })
+    });
+
+    // Emit a suggestion to use the type arg directly
+    // if there's a generic param with a matching name
+    // otherwise suggest removal of type binding
```

Comment:
```suggestion
    // Emit a suggestion to use turn the assoc item binding into a generic arg
    // if the relevant item has a generic param whose name matches the binding name
    // otherwise suggest the removal of the binding.
```